File: apps/portal/js/notification.js
/**
 *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 *  WSO2 Inc. licenses this file to you under the Apache License,
 *  Version 2.0 (the "License"); you may not use this file except
 *  in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 *
 */
var ApiUrl = "https://localhost:9443/notification/services/notifications/notificationApi";
var CREATE_PROXY = 'apis/login';
var UUID = "";
var notification = {
    "notification": {
        "notificationId": "33",
        "title": "invitation",
        "message": "its an invitation to update the vresion",
        "directUrl": "https://localhost:9443/portal",
        "users": ["admin@carbon.super"],
        "roles": ["admin"]
    }
};

login();
function login() {
    $.ajax({
        url: ues.utils.relativePrefix() + CREATE_PROXY,
        method: "GET",
        contentType: "text/plain",
        async: false,
        success: function (data) {
            UUID = data;
            console.log(UUID);
        },
        error: function (error) {
            console.log(error);
        }
    });
}


$('#notification').click(function () {
    $('#right-sidebar').slideToggle();
    //updateNotificationList();
    displayNotifications();
    /* $.ajax({
     url: 'https://localhost:9443/notification/services/notifications/notificationApi/notifications/' + '?uuid=' + UUID + '&username=' + userName + '&tenantId=-1234',
     method: 'POST',
     data: JSON.stringify(notification),
     async: false,
     contentType: "application/json",
     success: function (data) {
     console.log("notification added successfully" + data);

     },
     error: function (error) {
     console.log(error);
     }
     });*/
});

//display notifications in tha slide bar
function displayNotifications() {
    var notificationDetailsHbs = Handlebars.compile($('#all-notification-hbs').html());
    var details = getNotificationDetail();
    if (details) {
        $("#allNotifications").html(notificationDetailsHbs(details));
    }

}


/////get notification details for particular notification Id
function getNotificationDetail() {
    var detail;
    $.ajax({
        url: ApiUrl + '/notifications/detail/' + '?uuid=' + UUID + '&username=' + userName + '&tenantId=-1234',
        method: 'GET',
        async: false,
        contentType: "application/xml",
        success: function (response) {
            console.log("xxxxx");
            console.log(response.type);
            detail = (response);
        },
        error: function (error) {
            console.log("errorrrr");
            console.log(error);
        }
    });
    return detail;
}


function updateNotificationList() {
    $.ajax({
        url: ApiUrl + '/notifications/update/?notificationId=' + '1' + '&uuid=' + UUID + '&username=' + userName + '&tenantId=-1234',
        method: 'PUT',
        async: false,
        contentType: "text/plain",
        success: function (response) {
            console.log("xxxxx");
            console.log(response.type);

        },
        error: function (error) {
            console.log("errorrrr");
            console.log(error);
        }
    });
}

File: components/dashboard-notification/src/main/java/org/wso2/carbon/dashboard/notification/Notification.java
/*
*  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
*  WSO2 Inc. licenses this file to you under the Apache License,
*  Version 2.0 (the "License"); you may not use this file except
*  in compliance with the License.
*  You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing,
* software distributed under the License is distributed on an
* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
* KIND, either express or implied.  See the License for the
* specific language governing permissions and limitations
* under the License.
*
*/
package org.wso2.carbon.dashboard.notification;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.*;

@XmlRootElement(name = "notification")
public class Notification {
    private String notificationId;
    private String title;
    private String message;
    private String directUrl;
    private List<String> users;
    private List<String> roles;

    @XmlElement(name = "notificationId", required = true)
    public void setNotificationId(String notificationId) {
        this.notificationId = notificationId;
    }

    @XmlElement(name = "title", required = true)
    public void setTitle(String title) {
        this.title = title;
    }

    @XmlElement(name = "message", required = true)
    public void setMessage(String message) {
        this.message = message;
    }

    @XmlElement(name = "directUrl", required = true)
    public void setDirectUrl(String directUrl) {
        this.directUrl = directUrl;
    }

    @XmlElement(name = "users", required = false)
    public void setUsers(List<String> users) {
        this.users = users;
    }

    @XmlElement(name = "roles", required = false)
    public void setRoles(List<String> roles) {
        this.roles = roles;
    }

    public List<String> getRoles() {
        return roles;
    }

    public List<String> getUsers() {
        return users;
    }

    public String getNotificationId() {
        return notificationId;
    }

    public String getTitle() {
        return title;
    }

    public String getMessage() {
        return message;
    }

    public String getDirectUrl() {
        return directUrl;
    }
}
